src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public void register(String email, String password) {
+        User user = new User();
+        user.setLogin(email);
+        user.setPassword(password);
+        userService.add(user);
+    }
+
+    @Override
+    public User login(String login, String password) {
+        Optional<User> userFromDbOptional = userService.findByEmail(login);
+        if (userFromDbOptional.isEmpty()) {
+            throw new mate.academy.exception.AuthenticationException("Can't authenticate user");
+        }

```

Comment:
Checklist issue - Create only one condition for throwing AuthenticationException in login() method. You may combine two checks: whether the user has been found by login and does passwords match.

Also, import this Exception, don't write fully qualified class name